qnnpack hardswish - pytorch op integration

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36545 hardswish: make it work in static quantization
* **#36320 qnnpack hardswish - pytorch op integration**
* #36252 qnnpack hardswish - LUTs

Summary:

Hooks up the aten quantized hardswish op to the QNNPACK
path added in the previous PR.

Test Plan:

tests pass

will run benchmarking on mobile to confirm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20965043](https://our.internmc.facebook.com/intern/diff/D20965043)

vkuzo committed:
qnnpack hardswish - pytorch op integration

Summary:

Hooks up the aten quantized hardswish op to the QNNPACK
path added in the previous PR.

Test Plan:

tests pass

will run benchmarking on mobile to confirm

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on test/quantization/test_quantized.py:
```diff
@@ -369,30 +396,7 @@ def test_hardtanh(self, X, min_val, max_val):
-        torch.nn.quantized.functional.hardswish(qX_copy, inplace=True)
-        self.assertEqual(qX_copy, qY_hat,
-                         message="inplace Hardswish failed: {} vs {}".format(qY, qY_hat))
+        _test_hardswish(self, X, 'fbgemm')
```
Is there a reason why `test_x` is calling `_test_x`? Maybe rename this test as `test_x_fbgemm`? Or use a hard-coded parameter?

vkuzo replied:
it was a way to share the test code from the two different test classes, default and qnnpack (with different behaviors on what to skip per platform).  The test module is a part of the name during runs.  I'm flexible on if we can make it even better somehow - lmk!

z-a-f reviewed (approved):
LGTM

vkuzo committed:
Update on "qnnpack hardswish - pytorch op integration"


Summary:

Hooks up the aten quantized hardswish op to the QNNPACK
path added in the previous PR.

Test Plan:

tests pass

will run benchmarking on mobile to confirm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20965043](https://our.internmc.facebook.com/intern/diff/D20965043)

[ghstack-poisoned]

vkuzo committed:
fix macos test on "qnnpack hardswish - pytorch op integration"


Summary:

Hooks up the aten quantized hardswish op to the QNNPACK
path added in the previous PR.

Test Plan:

tests pass

will run benchmarking on mobile to confirm

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20965043](https://our.internmc.facebook.com/intern/diff/D20965043)

[ghstack-poisoned]
